sandbox/libs/authn/src/main/java/org/opensearch/authn/PermissionHandler.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import org.opensearch.authn.internal.InternalSubject;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Objects;
+
+public class PermissionHandler {
```

Comment:
It would help to start with an interface to address what are the inputs and outputs of this system.  With the current implementation it isn't as clear if the subject controls the data store or if it is managed by a 'swappable' system.  I think the latter would be more flexible in the long term.

Reply:
That makes sense. I will make an interface and then extend it.